Tests/QueryBuilderTests/MySqlQueryBuilderTest.php

```diff
@@ -0,0 +1,230 @@
+
+        $queryBuilder->select('words', ['id']);
+        $queryBuilder->reset();
+
+        $query = $queryBuilder->getSql();
+        $this->assertEmpty($query);
+    }
+
+    public function testSelect(): void
+    {
+        $queryBuilder = new MySqlQueryBuilder();
+
+        $query = $queryBuilder
+            ->select('words', ['id'])
+            ->getSql();
+
+        $this->assertEquals('SELECT id FROM words;', $query);
+    }
+
+    public function testInsert(): void
+    {
+        $queryBuilder = new MySqlQueryBuilder();
+
+        $query = $queryBuilder
+            ->insert('words', ['id', 'text'])
+            ->getSql();
+
+        $this->assertEquals('INSERT INTO words (id, text);', $query);
+    }
+
+    public function testWhereException(): void
+    {
+        $queryBuilder = new MySqlQueryBuilder();
+
+        $this->expectException(\Exception::class);
+
+        $queryBuilder
+            ->insert('words', ['id', 'text'])
+            ->where('words', '?')
+            ->getSql();
+    }
+
+    #[DataProvider('provideWhereSymbolConditions')]
+    public function testWhereConditionsWithSymbols(string $expectedResult, $condition): void
+    {
+        $queryBuilder = new MySqlQueryBuilder();
+
+        $query = $queryBuilder
+            ->select('words', ['id'])
+            ->where('words.text', $condition)
+            ->getSql();
+
+        $this->assertEquals($expectedResult, $query);
+    }
+
+    public static function provideWhereSymbolConditions(): array
+    {
+        return [
+            [
+                'SELECT id FROM words WHERE words.text = ?;',
+                '?'
+            ],
+            [
+                'SELECT id FROM words WHERE words.text = :word;',
+                ':word'
+            ],
+            [
+                "SELECT id FROM words WHERE words.text = 'word';",
+                'word'
+            ]
+        ];
+    }
+
+    #[DataProvider('provideWhereSymbolOperatorConditions')]
+    public function testWhereConditionsWithOperators(string $expectedResult, $condition, $operator): void
+    {
+        $queryBuilder = new MySqlQueryBuilder();
+
+        $query = $queryBuilder
+            ->select('words', ['id'])
+            ->where('words.text', $condition, $operator)
+            ->getSql();
+
+        $this->assertEquals($expectedResult, $query);
+    }
+
+    public static function provideWhereSymbolOperatorConditions(): array
+    {
+        return [
+            [
+                'SELECT id FROM words WHERE words.text IN (?, ?);',
+                '?, ?',
+                'IN'
+            ],
+            [
+                'SELECT id FROM words WHERE words.text IS NULL;',
+                ' ',
+                'IS NULL'
+            ],
+            [
+                'SELECT id FROM words WHERE words.text IS NOT NULL;',
+                ' ',
+                'IS NOT NULL'
+            ]
+        ];
+    }
+
+    public function testValuesExpectException(): void
```

Comment:
Would suggest for name of the test to be more descriptive. Something like (I do not know context so guessing) `testIfValuesCannotBeArray` or `testIfValuesFailsOnArray`.

Double check other methods too where naming could be improved.

Also some times it is okay to have testValues as you are testing a method but this case is different as you are testing case. Ideally this would be job for data provider as it provides tests with cases (different data). And my main suggestion would be to move this to a provider.